components/bl-signature-pad-component/component.json

```diff
@@ -0,0 +1,177 @@
+      "dataBinding": true
+    },
+    {
+      "name": "saveSignatureButton",
+      "label": "Save Signature Button Visibility",
+      "type": "checkbox",
+      "defaultValue": true,
+      "handlerId": "saveSignatureButtonVisibilityLogic",
+      "handlerLabel": "Save Signature Button Visibility Logic",
+      "handlerDescription": "This is a handler for the logic to determine if the Save Signature button should be visible. If the handler returns 'true' the button will be displayed."
+    },
+    {
+      "name": "clearButtonLabel",
+      "label": "Clear Button Label",
+      "defaultValue": "Clear",
+      "type": "text"
+    },
+    {
+      "name": "changeColorButtonLabel",
+      "label": "Change Color Button Label",
+      "type": "text",
+      "defaultValue": "Change Color"
+    },
+    {
+      "name": "undoButtonLabel",
+      "label": "Undo Button Label",
+      "type": "text",
+      "defaultValue": "Undo"
+    },
+    {
+      "name": "savePNGButtonLabel",
+      "label": "Save PNG Button Label",
+      "type": "text",
+      "defaultValue": "Save as PNG"
+    },
+    {
+      "name": "saveJPGButtonLabel",
+      "label": "Save JPG Button Label",
+      "type": "text",
+      "defaultValue": "Save as JPG"
+    },
+    {
+      "name": "saveSVGButtonLabel",
+      "label": "Save SVG Button Label",
+      "type": "text",
+      "defaultValue": "Save as SVG"
+    },
+    {
+      "name": "saveSignatureButtonLabel",
+      "label": "Save Signature Button Label",
+      "type": "text",
+      "defaultValue": "Save Signature"
+    }
+  ],
+  "eventHandlers": [
+    {
+      "name": "onClearClick",
+      "label": "On Clear Click Event",
+      "handlerDescription": "This event is fired when the user clicks the mouse or taps the the Clear button"
+    },
+    {
+      "name": "onUndoClick",
+      "label": "On Undo Click Event",
+      "handlerDescription": "This event is fired when the user clicks the mouse or taps the the Undo button"
+    },
+    {
+      "name": "onChangeColorClick",
+      "label": "On Change Color Click Event",
+      "handlerDescription": "This event is fired when the user clicks the mouse or taps the Change Color button",
+      "contextBlocks": [
+        {
+          "id": "color",
+          "label": "Color"
+        }
+      ]
+    },
+    {
+      "name": "onSaveClick",
+      "label": "On Save Click Event",
+      "handlerDescription": "This event is fired when the user clicks the mouse or taps one of the Save button"
+    },
+    {
+      "name": "onSaveSignatureClick",
+      "label": "On Save Signature Click Event",
+      "handlerDescription": "This event is fired when the user clicks the mouse or taps Save Signature button",
+      "contextBlocks": [
+        {
+          "id": "signature",
+          "label": "Signature"
+        },
+        {
+          "id": "fileName",
+          "label": "File Name"
```

Comment:
That might be overhead, used can generate the file name himself.

Or it's used somehow? Not only as utils block to generate random file name string?

Reply:
It is not used elsewhere. But when we create a file using codeless from the blob , we have to specify the file name. If the file name is not random, the downloaded file will be replaced with the following.